specs/jsonschema-core.md

```diff
@@ -47,51 +84,29 @@ document are to be interpreted as defined in [RFC 8259][rfc8259].
 
 ## Overview
 
-This document proposes a new media type `application/schema+json` to identify a
-JSON Schema for describing JSON data. It also proposes a further optional media
-type, `application/schema-instance+json`, to provide additional integration
-features. JSON Schemas are themselves JSON documents. This, and related
-specifications, define keywords allowing authors to describe JSON data in
-several ways.
+A JSON Schema represents a set of constraints and annotations that are applied
+to a JSON value. These constraints and annotations are declared using
+"keywords". A JSON value is considered valid against a schema if, and only if,
+it satisfies the constraint defined by every keyword in that schema.
 
-JSON Schema uses keywords to assert constraints on JSON instances or annotate
-those instances with additional information. Additional keywords are used to
-apply assertions and annotations to more complex JSON data structures, or based
-on some sort of condition.
-
-To facilitate re-use, keywords can be organized into vocabularies. A vocabulary
-consists of a list of keywords, together with their syntax and semantics. A
-dialect is defined as a set of vocabularies and their required support
-identified in a meta-schema.
-
-JSON Schema can be extended either by defining additional vocabularies, or less
-formally by defining additional keywords outside of any vocabulary. Unrecognized
-individual keywords are not supported.
-
-This document defines a set of core keywords that MUST be supported by any
-implementation, and cannot be disabled. These keywords are each prefixed with a
-"$" character to emphasize their required nature. These keywords are essential
-to the functioning of the `application/schema+json` media type.
+Schema evaluation is a recursive process. Some keywords contain one or more
+subschemas. These keywords can be used to create complex constraints or to
+describe compound values like arrays and objects. For example, to describe a
+JSON object, a schema can use the `type` keyword to declare that the value MUST
+be an object, and the `properties` keyword to apply separate schemas to each of
+the object's properties. This allows for the evaluation of a complex JSON
+document using a uniform recursive algorithm.
```

Comment:
I wonder if it would be good to add a note that the presence of `properties` doesn't imply that the instance must be an object.

Reply:
I think that would be a reasonable thing to put in the definition of `properties`, but I think it's getting too deep in the details for the overview section.

However, it could make sense to cover the general principle that some keywords only apply when the instance is of a particular type. But, don't think we want to get too into architectural principles in the overview. It could balloon as we decide that other architectural principles should be mentioned as well.